### PR TITLE
gen: implement generic global initializer support for deployments

### PR DESCRIPTION
**Pull request by doudou**
This may be used for any library that need a one-time initialization
in main. So far, orogen had only support for Qt. I needed this for
instance for GStreamer.

This is tested against Qt4 and GStreamer.

### CODE REVIEW

File: lib/orogen/gen/deployment.rb [hunk 1]
```diff
@@ -65,6 +65,50 @@ def task(name, klass)
...
+            @available_global_cpp_initializers = {}
+
+            # Register the C++ code generation for a given global initializer
+            #
+            # The method registers the initializer on {Spec::Deployment} to declare
+            # it to the spec loaders
+            #
+            # @param [Symbol] key the generator name, identical to the name used
+            #   in {Spec::TaskContext#needs_global_initializer}
+            def self.register_global_initializer(key,
```
Comment (g-arjones) [inline lib/orogen/gen/deployment.rb]: May we have an API that allows to pass a path to a file containing the initialiser? That would:

1. Cleanup the orogen spec a little;
2. Allow to have tests for the initializer

**Comment by g-arjones** [comment]
No tests?

**Comment by g-arjones** [comment]
> No tests?

Just found them :P

File: lib/orogen/gen/deployment.rb (hunk 1 above)
Comment (doudou) [reply lib/orogen/gen/deployment.rb]: No need for an API. If one wants to reuse definitions he/she can:
- make it available in RUBYLIB and require it. 
- define it in a base orogen project that is imported by others (e.g. if there is some common base task context or types that are subclassed)

**Pull request merged**